client/my-sites/themes/theme-showcase.jsx

```diff
@@ -47,6 +48,18 @@ import RecommendedThemes from './recommended-themes';
  */
 import './theme-showcase.scss';
 
+function getInstallThemeSlug( siteSlug, canUploadThemesOrPlugins ) {
+	if ( ! siteSlug ) {
+		return '/themes/upload';
+	}
+
+	if ( canUploadThemesOrPlugins ) {
+		return `https://${ siteSlug }/wp-admin/theme-install.php`;
+	}
```

Comment:
I found weird that after clicking on the "Install theme" button on a screen (Calypso) that is listing some themes, I'm taking away to a screen (WP Admin) that is again listing some themes where I need to click once more on the "Upload theme" button. 

Wouldn't that confuse users? Is there any way that the can automatically display the WP Admin theme upload form when users land on `theme-install.php`?

Reply:
Doesn't seem possible. The form is hidden by default, and it only shows up when clicking on the button: https://github.com/WordPress/WordPress/blob/51fe01e66de08e16e734b88af163ef80f4c4482e/wp-admin/js/theme.js#L1876-L1886

I guess we don't have any other alternative then.